bindings/rust/bench/src/openssl.rs

```diff
@@ -0,0 +1,110 @@
+        } {
+            Ok(_) => Ok(()),
+            Err(err) => {
+                if err.code() != ErrorCode::WANT_READ {
+                    Err(err.into())
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    }
+}
+
+impl TlsBenchHarness for OpenSslHarness {
+    fn new(crypto_config: &CryptoConfig) -> Result<Self, Box<dyn Error>> {
+        let client_buf = ConnectedBuffer::new();
+        let server_buf = client_buf.clone_inverse();
+
+        let cipher_suite = match crypto_config.cipher_suite {
+            CipherSuite::AES_128_GCM_SHA256 => "TLS_AES_128_GCM_SHA256",
+            CipherSuite::AES_256_GCM_SHA384 => "TLS_AES_256_GCM_SHA384",
+        };
+
+        let ec_key = match crypto_config.ec_group {
+            ECGroup::SECP256R1 => "P-256",
+            ECGroup::X25519 => "X25519",
+        };
+
+        let mut client_builder = SslConnector::builder(SslMethod::tls())?;
+        client_builder.set_ca_file(CA_CERT_PATH)?;
+        client_builder.set_min_proto_version(Some(SslVersion::TLS1_3))?;
+        client_builder.set_ciphersuites(cipher_suite)?;
+        client_builder.set_groups_list(ec_key)?;
+
+        let mut server_builder = SslAcceptor::mozilla_modern_v5(SslMethod::tls())?;
+        server_builder.set_certificate_chain_file(SERVER_CERT_CHAIN_PATH)?;
+        server_builder.set_private_key_file(SERVER_KEY_PATH, SslFiletype::PEM)?;
+        server_builder.set_min_proto_version(Some(SslVersion::TLS1_3))?;
+        server_builder.set_ciphersuites(cipher_suite)?;
+        server_builder.set_groups_list(ec_key)?;
+
+        let client_config = client_builder.build().into_context();
+        let server_config = server_builder.build().into_context();
+
+        let client_conn = SslStream::new(Ssl::new(&client_config)?, client_buf)?;
+        let server_conn = SslStream::new(Ssl::new(&server_config)?, server_buf)?;
+
+        Ok(Self {
+            client_conn,
+            server_conn,
+        })
+    }
+
+    fn handshake(&mut self) -> Result<(), Box<dyn Error>> {
+        for _ in 0..2 {
+            self.handshake_conn(Mode::Client)?;
+            self.handshake_conn(Mode::Server)?;
+        }
+        Ok(())
+    }
+
+    fn handshake_completed(&self) -> bool {
+        self.client_conn.ssl().is_init_finished() && self.server_conn.ssl().is_init_finished()
+    }
+
+    fn get_negotiated_cipher_suite(&self) -> CipherSuite {
+        match self
+            .client_conn
+            .ssl()
+            .current_cipher()
+            .expect("Handshake not completed")
+            .name()
+        {
+            "TLS_AES_128_GCM_SHA256" => CipherSuite::AES_128_GCM_SHA256,
+            "TLS_AES_256_GCM_SHA384" => CipherSuite::AES_256_GCM_SHA384,
+            _ => panic!("Unknown cipher suite"),
+        }
+    }
+
+    fn negotiated_tls13(&self) -> bool {
+        self.client_conn
+            .ssl()
+            .version2()
```

Comment:
Why `version2`?

Reply:
`version2()` corresponds to `SSL_version()` in the C API, but in the rust bindings `version()` is deprecated and is now `version_str()`, while `version2()` returns an `Option<SslVersion>`.

Reply:
😑 

I don't love that naming convention, but I guess we don't have much of a choice. It might be nice to add a comment along the lines of "`version()` is deprecated in favor of `version2()` which returns an enum instead of a string."

Reply:
Adding the comment to explain it